pytorch_lightning/core/lightning.py

```diff
@@ -957,6 +957,30 @@ def init_ddp_connection(self, global_rank: int, world_size: int, is_slurm_managi
         log.info(f"initializing ddp: GLOBAL_RANK: {global_rank}, MEMBER: {global_rank+1}/{world_size}")
         torch_distrib.init_process_group(torch_backend, rank=global_rank, world_size=world_size)
 
+    def configure_sync_bn(self, model: 'LightningModule', backend: str = 'torch') -> 'LightningModule':
+        """
+        Add global batchnorm for a model spread across multiple GPUs and nodes.
+
+        Override to synchronize batchnorm between specific process groups instead
+        of the whole world.
+
+        Args:
+            model: pointer to current :class:`LightningModule`.
+            backend: select between 'torch' and 'apex'.
+
+        Return:
+            LightningModule with batchnorm layers converted to sync batchnorm
+        """
+        # process_group = None defaults to whole world in both cases
+        if backend == 'torch':
```

Comment:
What's the difference of these two backends? I.e. if they're the same, should we really add another dependency on apex? Thought that we try to remove it from dependencies, now that pt has also native amp support (of course being backwards compatible with amp, but not introduce a new dependency with a new feature)

Reply:
There were a few issues with pytorch's implementation and hence the authors of swav were using apex for this. Also, they found apex to be faster.